src/main/java/tools/jackson/databind/introspect/AnnotatedCreatorCollector.java

```diff
@@ -66,24 +66,15 @@ Creators collect(Class<?> primaryMixIn)
          * ignorable after all annotations have been properly collapsed.
          */
         // AnnotationIntrospector is null if annotations not enabled; if so, can skip:
-        if (_collectAnnotations) {
-            if (_defaultConstructor != null) {
-                if (_intr.hasIgnoreMarker(_config, _defaultConstructor)) {
-                    _defaultConstructor = null;
-                }
-            }
-            // count down to allow safe removal
-            for (int i = constructors.size(); --i >= 0; ) {
-                if (_intr.hasIgnoreMarker(_config, constructors.get(i))) {
-                    constructors.remove(i);
-                }
-            }
-            for (int i = factories.size(); --i >= 0; ) {
-                if (_intr.hasIgnoreMarker(_config, factories.get(i))) {
-                    factories.remove(i);
-                }
-            }
+        if (_collectAnnotations && _defaultConstructor != null && _intr.hasIgnoreMarker(_config, _defaultConstructor)) {
+            _defaultConstructor = null;
         }
+
+        constructors.removeIf(constructor ->
```

Comment:
`removeIf` should perform better than `remove` in a loop - the `ArrayList` implementation collects a bitmask of elements to remove and then does it in a single copy, rather than repeated shifting

Reply:
Sounds good -- I can merge this separately?

Reply:
Sounds good. There is a slight behavior change here too - since some methods are not accessible, even if we try to `setAccessible`, the `null` filter is now unconditional whether we are looking for `@JsonIgnore` or not.